### DISCUSSION TITLE
Support match application name with title instead of classes.

### DISCUSSION THREAD

**Discussion by zw963** [xremap/xremap#641] [Ideas]
Given the document like following:

```markdown
application

application can be used for both modmap and keymap, which allows you to specify application-specific remapping.

application:
  not: Application
  # or
  not: [Application, ...]
  # or
  only: Application
  # or
  only: [Application, ...]

The application name can be specified as a normal string to exactly match the name, or a regex surrounded by /s like /application/.

To check the application names, you can use the following commands:
```

It's said, application name can match the name use regexp, but infact, this is only match on application classes. 

as example in the readme.

```
busctl --user call org.gnome.Shell /com/k0kubun/Xremap com.k0kubun.Xremap WMClasses
```

Can we add optional option, to allow support for match on application name too? because use name with regexp is more useful.

e.g.  following is the gnome output for bcompare

```
  {
    "in_current_workspace": true,
    "wm_class": "Beyond Compare",
    "wm_class_instance": "bcompare5",
    "title": "crystal-china <--> website - Folder Compare - Beyond Compare",
    "pid": 368028,
    "id": 1515430224,
    "frame_type": 0,
    "window_type": 0,
    "focus": false
  },
```

The class always is "Beyond Compare", but, the title of folder compare,  always match `/Folder Compare - Beyond Compare/`, but match on the title is not working for current xremap, can we any plan add this support?

Thanks.

**Comment by k0kubun** [comment]
We do support it for some platforms with `window:`. We probably don't support it for GNOME yet. We have closed https://github.com/xremap/xremap/pull/98, but I think we can port that patch to the current version. I would merge such a PR.

**Comment by hpccc53** [comment]
I think the requested feature is implemented. Otherwise please reopen this discussion.

**Reply by zw963** [reply]
~~I don't see the corresponding PR. Is this feature available in 0.14.10?~~

It's works on 0.14.10! thank you.
